### ISSUE TITLE
Automatic light/dark mode

### ISSUE THREAD

**Issue by yuejon** [zwave-js/zwave-js-ui#3607]
**Is your feature request related to a problem? Please describe.**
Currently, the dark mode toggle is locked in under the UI settings of the control panel. It used to be based on the selection at the login screen, but once logged in, it applies the setting that is saved under the user profile. Almost seems as if the toggle at the login screen is pointless since the user profile overrides it after getting logged in. I understand some people may want to have permanent dark mode but in my use case (and other folks perhaps), I use light mode when viewing the control panel on my desktop computer and dark mode on my phone.

**Describe the solution you'd like**
Add an option to automatically select light/dark mode based on system settings instead of permanently locking to one or the other.

**Comment by robertsLando** [comment]
> Add an option to automatically select light/dark mode based on system settings instead of permanently locking to one or the other.

This is the default behaviour when no mode is selected. Could you try to delete the preference from your local storage and settings?

**Comment by yuejon** [comment]
Is there a specific file I should be looking for to delete? I found settings.json and deleted the darkMode line but that didn't do anything.

**Comment by robertsLando** [comment]
@yuejon  After that you should restart your applicaction, also you should be sure nothing is in your localstorage

**Comment by yuejon** [comment]
I'm using zwave-js-ui as a docker container. Which localstorage are you referring to? I've recreated the container and it did not make a difference. I'm not exactly comfortable with removing all my data in the container's volume (like backups etc.):

![image](https://github.com/zwave-js/zwave-js-ui/assets/16492473/44b67dab-b573-45f0-9d6b-d93d9ba804e3)

**Comment by robertsLando** [comment]
localstorage is a browser thing, you don't have to remove anythign from there. Anyway I need to speak with other users in order to understand if they prefer the actual usage or this

**Comment by yuejon** [comment]
Ah got it. I tried incognito mode on my browsers instead of clearing the cache and it seemed to stay on light mode without switching to dark mode on my phone.

**Comment by joneshf** [comment]
It seems like the current automatic behavior only works when LocalStorage is cleared. After that, the setting stays at whatever it was at the time. That works fine if you always keep your system on the same theme. But many folks have their system theme change throughout the day (e.g. light during the day, dark during the night), and the current implementation of theming in zwave-js-ui doesn't support that behavior.

What most sites do these days is provide three options instead of two: dark, light, and system. The way the options work is:
- The "dark" option always sets the site theme to dark. This is the current behavior of enabling the dark mode toggle.
- The "light" option always sets the site theme to light. This is the current behavior of disabling the dark mode toggle.
- The "system" option follows whatever the system theme is in real time. This would be new behavior.

Would you be up for adding support for a third option of "system" to the UI? Or would you accept a PR to implement it if you don't have the bandwidth right now? It would probably be best to implement it as a radio or a dropdown (I don't see any radios in the settings now, so probably best to use a dropdown for consistency) since they're three mutually exclusive options. And maybe call it `colorScheme` instead of `darkMode`?

**Comment by robertsLando** [comment]
@joneshf Yeah, there was a user that started a PR but never finished it. I accept a PR if you would like to contribute :)

**Comment by joneshf** [comment]
Cool!

I took a look at the code base and it seems like one implementation might be:
- Change [`ui.darkMode`](https://github.com/zwave-js/zwave-js-ui/blob/d0efbb105cccd5e4faa875b40251e7240bda1a95/src/stores/base.js#L144) from a `boolean` to `ui.colorScheme` with type:
    ```TypeScript
    type ColorScheme = 'dark' | 'light' | 'system';
    ```
- Add a new value to the state:
    ```TypeScript
    type State = {
      …
      uiState: {
        darkMode: boolean;
        prefersColorSchemeDark: MediaQueryList;
      };
      …
    };
    ```

    The idea here is that `uiState.darkMode` is what Vuetify will use for its [`dark`](https://github.com/zwave-js/zwave-js-ui/blob/d0efbb105cccd5e4faa875b40251e7240bda1a95/src/App.vue#L2) property. The `uiState.prefersColorSchemeDark` is used to listen for changes and update `uiState.darkMode` when appropriate.
- Setup an event listener on `uiState.prefersColorSchemeDark` in the [`init`](https://github.com/zwave-js/zwave-js-ui/blob/d0efbb105cccd5e4faa875b40251e7240bda1a95/src/stores/base.js#L554-L583). In this event listener:
    - Do nothing if the `ui.colorScheme` is `'dark'` or `'light'`.
    - Otherwise, set the `uiState.darkMode` based on whether the `uiState.prefersColorSchemeDark` matches.
- Add a new `actions.setColorScheme` that sets `ui.colorScheme`, and updates `uiState.darkMode` appropriately. This should replace the current uses of [`actions.setDarkMode`](https://github.com/zwave-js/zwave-js-ui/blob/d0efbb105cccd5e4faa875b40251e7240bda1a95/src/stores/base.js#L584-L598).
- Change the Login component:
    - Change [`computed.darkMode`](https://github.com/zwave-js/zwave-js-ui/blob/d0efbb105cccd5e4faa875b40251e7240bda1a95/src/views/Login.vue#L126) to `computed.colorScheme` using `ui.colorScheme`.
    - Change [`computed.internalDarkMode`](https://github.com/zwave-js/zwave-js-ui/blob/d0efbb105cccd5e4faa875b40251e7240bda1a95/src/views/Login.vue#L128-L135) to `computed.internalColorScheme` using the new `colorScheme` and the base `actions.setColorScheme`.
    - Change the [checkbox](https://github.com/zwave-js/zwave-js-ui/blob/d0efbb105cccd5e4faa875b40251e7240bda1a95/src/views/Login.vue#L52-L58) to a `select` with the values of `ColorScheme`.
- Do a similar set of changes for Settings component as done for the Login component.
- Massage any of the other stuff.

Does that seem about right? Or am I missing something big?

Also, I _think_ we can set it up so that the current selection for dark/light is migrated over to this new setup. That way it can be transparent when folks upgrade.

**Comment by joneshf** [comment]
Went a slightly different way and tossed up https://github.com/zwave-js/zwave-js-ui/pull/4266. Let me know what you think whenever you get a minute!

**Comment by robertsLando** [comment]
@joneshf thanks for that! Well done, just did my review
